### PR TITLE
Add self-registration module.

### PR DESCRIPTION
**Pull request by yhontyk**
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [x] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [x] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

**Review by MichalMaler — APPROVED** [review]

**Review by azatsarynnyy — COMMENTED** [review]
My review was requested here, but I have no knowledge about "self-registration". Sorry.

File: modules/end-user-guide/partials/proc_logging-in-to-che-on-openshift-for-the-first-time-registering-as-a-new-user.adoc [hunk 1]
```diff
@@ -7,6 +7,11 @@
 
 This section describes how to log in to {prod-short} on OpenShift for the first time registering as a new user.
 
+[NOTE]
+====
+For {prod-short} installed without OpenShift OAuth support, self-registration of users is disabled by default, therefore the option to register a new user is not available on the login page. See xref:administration-guide:configuring-authorization.adoc#enabling-self-registration_{context}[].
```
Comment (themr0c) [inline modules/end-user-guide/partials/proc_logging-in-to-che-on-openshift-for-the-first-time-registering-as-a-new-user.adoc]: Could it be a prerequisite instead: 

* Self-registration is enabled. See xref:administration-guide:configuring-authorization.adoc#enabling-self-registration_{context}[].

File: modules/administration-guide/partials/proc_enabling-self-registration.adoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// configuring-authorization
+
+[id="enabling-self-registration_{context}"]
+= Enabling self-registration
+
+For {prod-short} installed without OpenShift OAuth support, self-registration of users is disabled by default, therefore the option to register a new user is not available on the login page. 
+
+.Procedure
+
+To enable Self-registration of users:
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_enabling-self-registration.adoc]:
```suggestion
To enable self-registration of users:
```

File: modules/end-user-guide/partials/proc_logging-in-to-che-on-openshift-for-the-first-time-registering-as-a-new-user.adoc (hunk 1 above)
Comment (yhontyk) [reply modules/end-user-guide/partials/proc_logging-in-to-che-on-openshift-for-the-first-time-registering-as-a-new-user.adoc]: Moved

**Comment by yhontyk** [comment]
@dmytro-ndp hi! Could you take a look when you have time at this PR? It's referenced in https://issues.redhat.com/browse/RHDEVDOCS-2394
I'm not sure if the QE team needs to test this scenario though.

**Review by themr0c — APPROVED** [review]

**Comment by themr0c** [comment]
Waiting for SME review. /cc @l0rd

File: modules/administration-guide/partials/proc_enabling-self-registration.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+[id="enabling-self-registration_{context}"]
+= Enabling self-registration
+
+For {prod-short} installed without OpenShift OAuth support, self-registration of users is disabled by default, therefore the option to register a new user is not available on the login page. 
+
+.Procedure
+
+To enable self-registration of users:
+
+. Navigate to the *Realm Settings* menu on the left and open the *Login* tab.
```
Comment (skabashnyuk) [inline modules/administration-guide/partials/proc_enabling-self-registration.adoc:18]: Should we emphasize here that it has to be admin? Should do that in Keycloak? Maybe add a link on how to get a keycloak password (if that documentation is available.)?

**Review by skabashnyuk — APPROVED** [review]
In general, the solution is ok. Tested on real OpenShift cluster with disabled **OAuth2**

File: modules/administration-guide/partials/proc_enabling-self-registration.adoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// configuring-authorization
+
+[id="enabling-self-registration_{context}"]
+= Enabling self-registration
```
Comment (azatsarynnyy) [inline modules/administration-guide/partials/proc_enabling-self-registration.adoc:4]: Is there a topic in the docs that would explain to me what `self-registration` is?
If that's something that a person, who is reading this doc, should already know - that's okay.
But as a reviewer and not being an expert in an administration part - I'm not sure I completely understand what is `self-registration`.

File: modules/administration-guide/partials/proc_enabling-self-registration.adoc (hunk 3 above)
Comment (yhontyk) [reply modules/administration-guide/partials/proc_enabling-self-registration.adoc:18]: @skabashnyuk I've added a prerequisite that mentions that you must be logged in as an admin - https://github.com/eclipse/che-docs/pull/1819/commits/305535f261b7ee9761ee1dd0c2886ad3e3d40973

File: modules/administration-guide/partials/proc_enabling-self-registration.adoc (hunk 4 above)
Comment (yhontyk) [reply modules/administration-guide/partials/proc_enabling-self-registration.adoc:4]: @azatsarynnyy I've added a few sentences on the introduction that explains what self-registration is used for https://github.com/eclipse/che-docs/pull/1819/commits/305535f261b7ee9761ee1dd0c2886ad3e3d40973 PTAL

**Pull request merged**